### ISSUE TITLE
When using the positional $ operator to update an element of array, value is casted to array

### ISSUE THREAD

**Issue by wikrsh** [Automattic/mongoose#5628]
**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**
When updating an element in array using the positional $ operator, passed value will be casted to Array.
In the script below, `elems` field will be updated to `[["c"], "b"]` from `["a", "b"]`.

**If the current behavior is a bug, please provide the steps to reproduce.**

```js
const mongoose = require('mongoose');

mongoose.set('debug', true);

const Schema = new mongoose.Schema({
  elems: [String]
});

const Model = mongoose.model('Test', Schema);

const main = async () => {
  await mongoose.connect('mongodb://localhost/test');

  const doc = await Model.create({elems: ['a', 'b']});
  const newDoc = await Model.findOneAndUpdate({
    _id: doc._id,
    elems: 'a'
  }, {
    $set: {
      "elems.$": 'c'
    }
  }, {
    new: true, upsert: false
  });
  // -- mongoose@4.11.10 --
  // Mongoose: tests.findAndModify({ elems: 'a', _id: ObjectId("59b11adf79133519770c23a9") }, [], { '$set': { 'elems.$': [ 'c' ] } }, { new: true, upsert: false, remove: false, fields: {} })

  // -- mongoose@4.11.1 --
  // Mongoose: tests.findAndModify({ elems: 'a', _id: ObjectId("59b11a4177f861195e97c0d7") }, [], { '$set': { 'elems.$': 'c' } }, { new: true, upsert: false, remove: false, fields: {} })

};

main()
  .then(() => process.exit())
  .catch((err) => console.error(err));
```

**What is the expected behavior?**
In the script above, the expected behavior would be updating `elems` field to `["c", "b"]`.
In mongoose@4.11.1, I get the expected result `["c", "b"]`. But in mongoose@4.11.10, I get `[["c"], "b"]`.

**Please mention your node.js, mongoose and MongoDB version.**
node.js : 8.4.0
mongoose : 4.11.10
MongoDB : 3.4.3

**Comment by varunjayaraman** [comment]
thanks for the full repro script, this looks like a bug for sure

**Comment by vkarpov15** [comment]
Thanks for the detailed repro, fix will be in 4.11.13
